### PR TITLE
REST API: Install plugin translations after the plugin install. 

### PR DESCRIPTION
**Pull request by dd32**
REST API: Install plugin translations after the plugin install. This only installs for the plugin in question, not all plugins.

Support for retrieving the langauge pack alongside the install API request was added in https://meta.trac.wordpress.org/changeset/10091 to avoid having to make a plugin update check during the REST API check.

Trac ticket:  https://core.trac.wordpress.org/ticket/50732

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/rest-api/endpoints/class-wp-rest-plugins-controller.php [hunk 1]
```diff
@@ -355,6 +356,14 @@ public function create_item( $request ) {
 			}
 		}
 
+		// Install translations.
+		if ( ! empty( $api->language_pack ) ) {
+			$lp_upgrader = new Language_Pack_Upgrader( $skin );
+
+			// Assume a singular language pack was returned.
```
Comment (ocean90) [inline src/wp-includes/rest-api/endpoints/class-wp-rest-plugins-controller.php:363]: How would this work with plugins like [Preferred Languages](https://de.wordpress.org/plugins/preferred-languages/) or sites with content in multiple languages? Currently not sure if blocks with server-side rendering are allowed.

Comment (dd32) [reply src/wp-includes/rest-api/endpoints/class-wp-rest-plugins-controller.php:363]: Good question, and is something I'd thought about, even when making this choice. The WordPress.org API will only return LPs here that match the locale of the API request, and that MUST be a singular locale.

Since WordPress still only supports a single locale being loaded, this only installs the language of the current request in order to speed things up.
If the plugin has extra languages in use by other users (or even the current user, through Preferred languages), then it'll be installed on the regular translation update schedule on the twice-daily cron or via Dashboard -> Updates.

The alternative was to run a plugins update check post-install, and then install every language update for core/themes/plugins,  or to run the update check and then filter the `wp_get_current_translation_updates()` response and only install translations relevant to the current plugin.

tl;dr: This is a limitation of the WordPress.org API's and the WordPress Translation update system. The API request that this is relying upon only supports a singular locale and can't support multiple locales easily in a coherent manner.

Comment (ocean90) [reply src/wp-includes/rest-api/endpoints/class-wp-rest-plugins-controller.php:363]: Another idea: Change `translations_api()` to support passing custom values for the `locale` parameter and let the API only return results based on the provided locale(s).

```
// See wp_update_plugins()
$locales = array_values( get_available_languages() );
$locales = apply_filters( 'plugins_update_check_locales', $locales );
$locales = array_unique( $locales );

$translations = translations_api(
	'plugins',
	[
		'slug'   => 'my-block-plugin',
		'locale' => implode( ',', $locales ),
	]
);

// Install translations…
```

Comment (dd32) [reply src/wp-includes/rest-api/endpoints/class-wp-rest-plugins-controller.php:363]: That could also work, although I'd be concerned about how long the extra API & multiple locales would add to the REST API call.

I don't see this as a 5.5 blocker for this PR, and would make sense for a future iteration.

Comment (dd32) [reply src/wp-includes/rest-api/endpoints/class-wp-rest-plugins-controller.php:363]: I should also mention, the `translations_api()` method currently does pass the current locale to the translations API, but it doesn't actually respect that parameter and just returns all language packs applicable for the plugin/version.

It would be possible for that API to be used currently with client-side filtering, or to add locale-filtering api-side, although that may change how `translations_api()` is being called by other code.

Comment (dd32) [reply src/wp-includes/rest-api/endpoints/class-wp-rest-plugins-controller.php:363]: Right, I now understand - It'd be like asking for `es_ES` and `es_MX` language packs and only installing `es_ES` because `es_MX` doesn't exist and having the UI still be translated (Of course, both might be installed).
That's a reasonable request, where as I was previously only concerned about a multi-user site where there might be `de_DE` and `es_ES` users.

Comment (dd32) [reply src/wp-includes/rest-api/endpoints/class-wp-rest-plugins-controller.php]: 4b4923b switches this around to install all language packs for the plugin that are applicable to the current site, and should work with the Preferred Languages plugin.

My concerns around speed are still applicable, however, are mitigated as I've avoided an extra API request and that for most sites, this will only ever have a singular matching pack (or perhaps two in the case of a language fallback). The benefit of proper translations far outweighs the potential extra API time.

**Comment by dd32** [comment]
Merged in beba4bb73a10b97809105b3184da7006fbe25f55
